### PR TITLE
Update SDK Version to 30

### PR DESCRIPTION
**Pull request by SeijiV13**
- updated native apple authentication to 2.0.2
- compileSdkVersion  = 30
- targetSdkVersion = 30
- minSdkVersion = 21
- buildToolsVersion = 30.0.2
- firebaseMessagingVersion = 21.1.0
- updated gradle to 6.8

### CODE REVIEW

**Review by wrandall22 — CHANGES_REQUESTED** [review]
Looks like you need to update GHA a bit for codecov to come through. See https://github.com/codecov/codecov-action/issues/190#issuecomment-790729633 as a possible solution.

**Comment by SeijiV13** [comment]
@wrandall22 updated fetch-depth to 2 for GHA

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -249,7 +249,7 @@ jobs:
       - uses: actions/checkout@v2
         with:
           # Used for computing the Android versionCode https://github.com/CruGlobal/missionhub-react-native/blob/4c37730c21baeea3a304702b1bbbbce3d11f320d/android/app/build.gradle#L145
-          fetch-depth: 0
+          fetch-depth: 2
```
Comment (wrandall22) [inline .github/workflows/ci-cd.yml:252]: I think this was supposed to change on the `test` job, not the `android-deploy`

Comment (SeijiV13) [reply .github/workflows/ci-cd.yml]: @wrandall22 my bad already added the fetch-depth to the test

**Review by wrandall22 — APPROVED** [review]

**Pull request merged**